dcpy/connectors/esri/arcgis_feature_service.py

```diff
@@ -12,41 +12,142 @@
 )
 import yaml
 
-from dcpy.models.connectors.esri import FeatureServer
+from dcpy.models.connectors.esri import FeatureServer, FeatureServerLayer
 import dcpy.models.product.dataset.metadata as models
 from dcpy.utils.logging import logger
 
 
-def get_metadata(dataset: FeatureServer) -> dict:
-    resp = requests.get(f"{dataset.url}", params={"f": "pjson"})
+def get_feature_server_metadata(feature_server: FeatureServer) -> dict:
+    """Given a FeatureServer, return its metadata as a dictionary"""
+    resp = requests.get(f"{feature_server.url}", params={"f": "pjson"})
     resp.raise_for_status()
     error = resp.json().get("error")  # 200 responses might contain error details
     if error:
         raise Exception(f"Error fetching ESRI Server metadata: {error}")
     return resp.json()
 
 
-def get_data_last_updated(dataset: FeatureServer) -> datetime:
-    metadata = get_metadata(dataset)
+def get_feature_server_layers(
+    feature_server: FeatureServer,
+) -> list[FeatureServerLayer]:
+    """Given a FeatureServer, look up and return its available layers"""
+    resp = get_feature_server_metadata(feature_server)
+    return [
+        FeatureServerLayer(
+            server=feature_server.server,
+            name=feature_server.name,
+            layer_name=l["name"],
+            layer_id=l["id"],
+        )
+        for l in resp["layers"]
+    ]
+
+
+def resolve_layer(
```

Comment:
I refactored this just slightly - seemed to be a bit cleaner with a `match ... case`. See latest commit